commands/start.js

```diff
@@ -96,16 +73,83 @@ function start(urc) {
 
     const renderedErrors = renderWebpackErrors(stats);
     if (renderedErrors) {
-      logger.error(renderedErrors);
+      logger.error(new Error(renderedErrors));
       return;
     }
 
+    logger.log('Compiled JS.');
+
     if (urc.stats) {
       writeWebpackStats(urc.stats, stats);
+      return;
     }
   };
 
-  compiler.watch({ ignored: [/node_modules/] }, onCompilation);
+  watcher.watch({ ignored: [/node_modules/] }, onCompilation);
+
+  const webpackFirstRun = new Promise(resolve => {
+    emitter.once('ready', stats => {
+      resolve(stats);
+      emitter = undefined;
```

Comment:
I wanted a way clean way to signal the first run, a simple promise resolving on first run seemed like a good fit. I am setting the `emitter` as undefined as a precautionary measure. 
As a way to make the flow a bit easier to understand
1. I can possibly remove setting it undefined, since it is anyway using `.once` and EventEmitter is smart enough to not broadcast if there are no event handlers attached. 
2. Keep it the way it was
3. Any other way you can think of to signal first run ?

Reply:
Ah, thanks makes it much simpler! Thanks @davidtheclark . 
Also, I was wondering about calling `onFirstRun` after the error checking https://github.com/mapbox/underreact/pull/20/files/a36d63d23df482f4e942bca9fd24ef6179570ab9#diff-641e74433c6e039886f6892940b07b5dR78, since there is no point in running other compilers before the error checking as the output bundle maybe not be there.

Reply:
👍